time_execution/decorator.py

```diff
@@ -25,15 +25,22 @@ def write_metric(name, **metric):
         backend.write(name, **metric)
 
 
-def _apply_hooks(**kwargs):
+def _apply_hooks(hooks=None, **kwargs):
+    hooks = hooks or settings.hooks
     metadata = dict()
-    for hook in settings.hooks:
+    for hook in hooks:
```

Comment:
This will override the configured global hooks. Which means we don't cover the case where you want to use the global hooks and also local/custom ones.
So, I suggest a different approach:
- default hooks (the global ones);
- hooks (local);
- another option (keyword argument) `disable_default_hooks=False`. When this parameter is `True`, only the local hooks will be used. When `False`, the hooks will be merged.

Reply:
Agree. I renamed param `hooks` to `extra_hooks` to specify that this hook will be merged with default and added flag to exclude default ones.